src/utils/markets.tsx

```diff
@@ -787,67 +795,56 @@ export function useAllOpenOrdersBalances() {
   return openOrdersBalances;
 }
 
-export function useAllOpenOrders(): {
+export const useAllOpenOrders = (): {
   openOrders: { orders: Order[]; marketAddress: string }[] | null | undefined;
   loaded: boolean;
   refreshOpenOrders: () => void;
-} {
+} => {
   const connection = useConnection();
-  const { connected } = useWallet();
-  const [
-    openOrdersAccounts,
-    openOrdersAccountsConnected,
-  ] = useAllOpenOrdersAccounts();
-  const [marketInfos, marketInfosConnected] = useAllMarkets();
-  const openOrdersAccountsByAddress: {
-    [marketAddress: string]: OpenOrders[];
-  } = {};
-  for (let account of openOrdersAccounts || []) {
-    const marketsAddr = account.market.toBase58();
-    if (!(marketsAddr in openOrdersAccountsByAddress)) {
-      openOrdersAccountsByAddress[marketsAddr] = [];
-    }
-    openOrdersAccountsByAddress[marketsAddr].push(account);
-  }
-  const marketsByAddress = Object.fromEntries(
-    (marketInfos || []).map((info) => [info.market.publicKey.toBase58(), info]),
-  );
-  const getAllOpenOrders = async () => {
-    return await Promise.all(
-      Object.keys(openOrdersAccountsByAddress).map(async (marketAddr) => {
-        const market = marketsByAddress[marketAddr].market;
-        const [bids, asks] = await Promise.all([
-          market.loadBids(connection),
-          market.loadAsks(connection),
-        ]);
-        return {
-          orders: market.filterForOpenOrders(
-            bids,
-            asks,
-            openOrdersAccountsByAddress[marketAddr],
-          ),
-          marketAddress: marketAddr,
-        };
-      }),
-    );
+  const { connected, wallet } = useWallet();
+  const [loaded, setLoaded] = useState(false);
+  const [refresh, setRefresh] = useState(0);
+  const [openOrders, setOpenOrders] = useState<
+    { orders: Order[]; marketAddress: string }[] | null | undefined
+  >(null);
+
+  const refreshOpenOrders = () => {
+    setRefresh((prev) => prev + 1);
   };
-  const cacheKey = tuple(
-    'getAllOpenOrders',
-    openOrdersAccountsConnected,
-    (openOrdersAccounts || []).length,
-    connection,
-    connected,
-    marketInfosConnected,
-  );
-  const [openOrders, loaded] = useAsyncData(getAllOpenOrders, cacheKey, {
-    refreshInterval: _VERY_SLOW_REFRESH_INTERVAL,
-  });
+
+  useEffect(() => {
+    if (connected) {
+      const getAllOpenOrders = async () => {
+        setLoaded(false);
+        const _openOrders: { orders: Order[]; marketAddress: string }[] = [];
+        for (let i = 0; i < USE_MARKETS.length; i++) {
+          const market = await Market.load(
+            connection,
+            USE_MARKETS[i].address,
+            undefined,
+            USE_MARKETS[i].programId,
+          );
+          const orders = await market.loadOrdersForOwner(
```

Comment:
Let's add a `cacheDurationMs` parameter here which will cut down on the number of times we call `getProgramAccounts` which is a non-performant query for validators.

Reply:
30 seconds seems reasonable to me.

Reply:
Done!